src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -86,7 +86,7 @@ public EETypeNode(NodeFactory factory, TypeDesc type)
         public override bool ShouldSkipEmittingObjectNode(NodeFactory factory)
         {
             // If there is a constructed version of this node in the graph, emit that instead
-                if (ConstructedEETypeNode.CreationAllowed(_type))
+            if (ConstructedEETypeNode.CreationAllowed(_type) || _type.IsCanonicalDefinitionType(CanonicalFormKind.Any))
```

Comment:
This change was suggested by @davidwrighton. I am not familiar with all the side effects associated with having to keep track of multiple NodeCaches for types.
@davidwrighton can you comment on this?